### PR TITLE
Default carrier name was displayed as 0

### PR DESCRIPTION
**Pull request by Gitdyr**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In the back office order details the carrier name was displayed as "0" if the default carrier was used
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |
| How to test?  | Install a shop from scratch and create an order where the default carrier is selected (the one bearing the shop's name)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12140)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
I could not reproduce your bug. Are you sure there is a `0` stored in database? 
If it is a data problem which can not be retrieve by sql request, I think it must be changed in PHP (instead of SQL request).

**Comment by Gitdyr** [comment]
Yes, there is a 0 in the database.

Notice the method replaceZeroByShopName() in classes/Cart.php that fixes the problem in the cart.

**Comment by Quetzacoalt91** [comment]
Well, even if this sound less optimized, I would rather see it as a for loop, applying the method on the `carrier_name` of each line.

This would make the code easier to read, and will avoid some potential security issues by applying a PHP variable in the SQL request.

Thanks

**Comment by Gitdyr** [comment]
Updated pull request.

File: classes/order/Order.php [hunk 1]
```diff
@@ -1919,6 +1919,12 @@ public function getShipping()
             WHERE o.`id_order` = ' . (int) $this->id . '
             GROUP BY c.id_carrier'
         );
+        foreach ($results as &$row) {
+            if ($row['carrier_name'] == '0') {
+                $row['carrier_name'] = Carrier::getCarrierNameFromShopName();
+            }
```
Comment (Quetzacoalt91) [inline classes/order/Order.php]: Prefer using directly the existing code.

```php
$row['carrier_name'] = Cart::replaceZeroByShopName($row['carrier_name'], null);
```

**Review by Quetzacoalt91 — CHANGES_REQUESTED** [review]

**Comment by Gitdyr** [comment]
Updated pull request.

**Review by PierreRambaud — DISMISSED** [review]
LGTM, wdyt @Quetzacoalt91 ?

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @Gitdyr

**Pull request merged**